### ISSUE TITLE
Some linter messages highlighting wrong word

### ISSUE THREAD

**Issue by Ldash4** [vgalaktionov/moonscript-vscode-extension#2]
This is probably best described with an image:
![image](https://user-images.githubusercontent.com/30183784/48628639-f7c57500-e9b7-11e8-9031-4ed23ff876d8.png)
The linter message in this case is "accessing global 'bit'".
It seems that when trying to find the word to highlight, it finds the first occurrence of the word, which isn't always correct.
I don't have experience with either this repository nor moonpick, so I can't tell what project this error stems from.

**Comment by vgalaktionov** [comment]
Yeah, that's most likely an issue with the wonky `getDiagnostics` function in §extension.ks. Currently it indeed finds the first occurrence of a word. 

I'm not currently using moonscript for any project so will likely not find time to work on this. Happy to take contributions though!

**Comment by vgalaktionov** [comment]
Improving the implementation of getDiagnostics will also fix #3  - right now the implementation is very naive.
